app/views/ReportView/components/CopyNumber/index.tsx

```diff
@@ -88,14 +89,32 @@ const CopyNumber = ({
             } of cnvsResp) {
               /* Show either RPKM or TPM columns based on which is populated */
               if (tpm !== null) {
-                setVisibleCols((prevVal) => [...prevVal, 'tpm']);
+                nextVisible.push('tpm');
                 break;
               }
               if (rpkm !== null) {
-                setVisibleCols((prevVal) => [...prevVal, 'rpkm']);
+                nextVisible.push('rpkm');
                 break;
               }
             }
+            /* Show either Normal or kIQR columns based on which is populated */
+            for (const {
```

Comment:
Some of these can have a lot of entries so it would be ideal to have the logic here in one loop instead of 2. Possibly something like:

```
            for (const {
              gene: {
                expressionVariants: {
                  tpm, rpkm, primarySiteFoldChange, primarySitekIQR
                },
              },
            } of cnvsResp) {
              /* Show either RPKM or TPM columns based on which is populated */
              if (tpm !== null && !nextVisible.includes('tpm')) {
                nextVisible.push('tpm');
              }
              if (rpkm !== null && !nextVisible.includes('rpkm')) {
                nextVisible.push('rpkm');
              }
              if (primarySiteFoldChange !== null && !nextVisible.includes('primarySiteFoldChange')) {
                nextVisible.push('primarySiteFoldChange');
              }
              if (primarySitekIQR !== null && !nextVisible.includes('primarySitekIQR')) {
                nextVisible.push('primarySitekIQR');
              }
              if (nextVisible.length === 2) {
                break;
              }
            }
```

Reply:
yeah this will work better